src/main/java/neqsim/processSimulation/processEquipment/compressor/Compressor.java

```diff
@@ -670,6 +670,44 @@ public void run(UUID id) {
     setCalculationIdentifier(id);
   }
 
+  /** {@inheritDoc} */
+  @Override
+  public void runTransient(double dt, UUID id) {
+    if (getCalculateSteadyState()) {
+      run(id);
+      increaseTime(dt);
+      return;
+    }
+
+    // runController(dt, id);
+    System.out.println("delta p compressor "
+        + (outStream.getThermoSystem().getPressure() - inStream.getThermoSystem().getPressure()));
+
+    double guessFlow = inStream.getThermoSystem().getFlowRate("m3/hr");
+    outStream.getThermoSystem().init(3);
+    inStream.getThermoSystem().init(3);
+    double polytropicHead = outStream.getThermoSystem().getEnthalpy("kJ/kg")
+        - inStream.getThermoSystem().getEnthalpy("kJ/kg");
+    System.out.println("polytropic head " + polytropicHead);
+    double actualFlowRateNew = 10.0;
+    actualFlowRateNew = getCompressorChart().getFlow(polytropicHead, getSpeed(), guessFlow);
+
+    try {
+      inStream.getThermoSystem().setTotalFlowRate(actualFlowRateNew, "Am3/hr");
+      inStream.getThermoSystem().init(3);
+      // inStream.run(id);
+    } catch (Exception e) {
+      logger.error(e.getMessage());
+    }
+    try {
+      outStream.getThermoSystem().setTotalFlowRate(actualFlowRateNew, "Am3/hr");
+      outStream.getThermoSystem().init(3);
+    } catch (Exception e) {
+      logger.error(e.getMessage());
```

Comment:
I noticed that this code segment uses a `try-catch` block to catch exceptions thrown by the methods of the ThermoSystem class.  It might be needed to silence exceptions in this case?

Just some suggestions:

1. Instead of catching all exceptions with catch (Exception e), consider catching only the specific exceptions that could be thrown by the setTotalFlowRate and init methods. This will make it easier to identify and handle errors that are specific to these methods.

2. Add validation tot the `actualFlowRateNew` to avoid causing exceptions. maybe it has to be not-null or positive?

3. Unless it is needed, it is good practice to throw the exceptions after logging it. This way it is less likely to cause issues for consumers of this code.

Reply:
good suggestings. I think all are good ideas, and we should look into implementing this.

Reply:
for the moment I work on getting the dynamic compressor to work, when it is working I will look into your suggestion.

Reply:
@jorgenengelsen I have removed the try catch exception, and added a check for actualFlowRateNew.

Reply:
It looks good to me!